### PR TITLE
cql library draft and versioning

### PR DESCRIPTION
**Pull request by nmorasb**
## MADiE PR

Jira Ticket: [MAT-4128](https://jira.cms.gov/browse/MAT-4128)
(Optional) Related Tickets:
 [MAT-4103](https://jira.cms.gov/browse/MAT-4103)

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        Version version =
+            cqlLibraryRepository
+                .findMaxMinorVersionByGroupIdAndVersionMajor(
+                    cqlLibrary.getGroupId(), cqlLibrary.getVersion().getMajor())
+                .orElse(new Version());
+        return version.toBuilder().minor(version.getMinor() + 1).build();
+      }
+    } catch (RuntimeException ex) {
+      log.error("VersionController::updateVersion Exception while updating version number", ex);
+      throw new RuntimeException("Unable to update version number", ex);
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java]: Would this be caught by any advice?

Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java]: Would this be caught by any advice?

**Review by adongare — DISMISSED** [review]
Looks great!

**Review by sb-prateekkeerthi — DISMISSED** [review]

File: src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java (hunk 1 above)
Comment (nmorasb) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java]: that was one of the codacy items. I've changed it to a new InternalServerErrorException (and it will be caught by the advice and returned as a 500 error)

**Review by adongare — APPROVED** [review]
